Retrospring/retrospring
# Integrate feedback forms with Canny

This integrates feedback and bug report forms with [Canny](https://canny.io)

Wasn't sure where to put the links so they are at least somewhat visible (but not a main feature), so I put them into the profile/user dropdown:
![image](https://user-images.githubusercontent.com/1774242/150631059-22003e3a-d548-428c-bf48-e3ff5e153ee0.png)

**Prerequisites**
* Get Canny credentials:
  * SSO token: https://developers.canny.io/install/widget/sso
  * Board tokens: https://developers.canny.io/install/widget/web
  * And put them into `config/justask.yml`

**Testing**
* Try accessing `/feedback/bugs`
  * This should redirect to `/feedback/consent`
* After consenting, try accessing `/feedback/consent`
  * This should redirect to `/feedback/bugs`
* On one of the form pages, check one of the script tags after the feedback form, including an `ssoToken`
* Create an example post in one of the boards
  * If using an admin account (determined by the used email) this will be directly visible
  * Otherwise the screen name of the user and the profile picture (which won't work since the URL is only local) should be set
* If the `canny` section of the comments is commented out, visiting a feedback path should redirect to `root_path`
